integration/backward_compatibility.go

```diff
@@ -11,20 +11,48 @@ var DefaultPreviousVersionImages = map[string]e2emimir.FlagMapper{
 		e2emimir.SetFlagMapper(map[string]string{"-ingester.ring.readiness-check-ring-health": "false"}),
 		e2emimir.RemoveFlagMapper([]string{"-ingester.native-histograms-ingestion-enabled"}),
 		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
 	),
 	"grafana/mimir:2.6.0": e2emimir.ChainFlagMappers(
 		e2emimir.RemoveFlagMapper([]string{"-ingester.native-histograms-ingestion-enabled"}),
 		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
+	),
+	"grafana/mimir:2.7.1": e2emimir.ChainFlagMappers(
+		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
+	),
+	"grafana/mimir:2.8.0": e2emimir.ChainFlagMappers(
+		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
+	),
+	"grafana/mimir:2.9.1": e2emimir.ChainFlagMappers(
+		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
+	),
+	"grafana/mimir:2.10.0": e2emimir.ChainFlagMappers(
+		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
+	),
+	"grafana/mimir:2.11.0": e2emimir.ChainFlagMappers(
+		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
+	),
+	"grafana/mimir:2.12.0": e2emimir.ChainFlagMappers(
+		removePartitionRingFlags,
+		removeRemoteReadLimitsFlags,
 	),
-	"grafana/mimir:2.7.1":  removePartitionRingFlags,
-	"grafana/mimir:2.8.0":  removePartitionRingFlags,
-	"grafana/mimir:2.9.1":  removePartitionRingFlags,
-	"grafana/mimir:2.10.0": removePartitionRingFlags,
-	"grafana/mimir:2.11.0": removePartitionRingFlags,
-	"grafana/mimir:2.12.0": removePartitionRingFlags,
 }
 
+// Always remove this flag so that it's not a pain to run GEM integration tests.
+// This temporarily flag will be removed soon from Mimir.
+var defaultPreviousVersionGlobalOverrides = removeRemoteReadLimitsFlags
```

Comment:
Note to reviewers: setting up overrides in GEM is a true pain. I want to avoid it. This trick allow us to always (forcefully) remove the new config option, so that we don't have to deal with it in GEM.